extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,261 @@
+    this.attachContent_();
+    this.initializeListeners_();
+    createShadowRootWithStyle(this.element, this.quizEl_, CSS);
+  }
+
+  /**
+   * Reacts to RTL state updates and triggers the UI for RTL.
+   *
+   * @param {boolean} rtlState
+   * @private
+   */
+  onRtlStateUpdate_(rtlState) {
+    this.mutateElement(() => {
+      rtlState
+        ? this.quizEl_.setAttribute('dir', 'rtl')
+        : this.quizEl_.removeAttribute('dir');
+    });
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === 'flex-item';
+  }
+
+  /**
+   * @return {?Element}
+   */
+  getQuizElement() {
+    return this.quizEl_;
+  }
+
+  /**
+   * Finds the prompt and options content
+   * and adds it to the quiz element.
+   *
+   * @private
+   */
+  attachContent_() {
+    // TODO(jackbsteinberg): Optional prompt behavior must be implemented here
+    const promptInput = this.element.children[0];
+    // First child must be heading h1-h3
+    if (!['h1', 'h2', 'h3'].includes(promptInput.tagName.toLowerCase())) {
+      dev().error(
+        TAG,
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    const prompt = document.createElement(promptInput.tagName);
+    prompt.textContent = promptInput.textContent;
+    prompt.classList.add('i-amp-story-quiz-prompt');
+    promptInput.remove();
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      dev().error(TAG, 'Improper number of options');
+    }
+
+    this.quizEl_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      dev().error(TAG, 'Too many children');
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the quiz element.
+   *
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const convertedOption = buildOptionTemplate(option);
+
+    // Fill in the answer choice
+    convertedOption.querySelector(
+      '.i-amp-story-quiz-answer-choice'
+    ).textContent = answerChoiceOptions[index];
+
+    // Transfer the option information into a span then remove the option
+    const optionText = document.createElement('span');
+    optionText.setAttribute('class', 'i-amp-story-quiz-option-text');
+    optionText.textContent = option.textContent;
+    convertedOption.append(optionText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    convertedOption.setAttribute(
+      'id',
+      `${this.element.id}+${answerChoiceOptions[index]}`
+    );
+
+    convertedOption.setAttribute(
+      'on',
+      `tap:${this.element.id}.triggerInteraction`
```

Comment:
Nit: should the name be a bit more user friendly like `answer` or `select` or anything you like?